README.md

```diff
@@ -308,3 +309,112 @@ compose(
+
+  - invalidation of all *dependents* will be automatic (all of the *downset* of any invalidated node is recursively invalidated)
+
+  - refetch of some *dependencies* may be transparent/automatic (if needed given the cache policies)
+
+To explain this better, let's be more precise about what "invalidating" means
+
+We can identify 2 phases:
+
+### invalidate
+
+which caches (cached fetches), e.g. upon executing a certain "command", should be invalidated?
+
+**It's explicit**: the invalidate phase, starting from a potentially (half) filled cache, should invalidate (aka delete) specific cached values that we know could now be obsolete. This overrides any caching policy, it just *deletes*.
+
+It is recursive: invalidating a node, the entire *downset* for that node is considered obsolete as well and thus all cached values (matching current input `A`) of *dependents* of that node are deleted as well.
+
+### refetch
+
+Which fetches should be re-run (`fetch()`ed) after the invalidation phase?
+
+**It's implicit**: the system knows exactly which are the `fetch` functions we want to re-run at any point in time, given it knows if someone is *observing* each cached fetch or not. Only observed fetches are re-`fetch()`ed; non-currently-observed ones will be `fetch()`ed when someone asks for them.
+
+To clarify these two phases and the `invalidate` api, here's an example:
+
+### Example
+
+Assume our `graph` is composed of 3 compound nodes, `A`, `B` and `C`.
+
+> **Note**: lower level `fetch`es are in general >= to the number of compound nodes, the number of compound nodes corresponds to the user-provided `fetch` functions. Here we reason in terms of compound nodes though.
+
+The graph is arranged in this way in terms of dependencies:
+
+```
+    A
+   / \
+  B   C
+```
+
+In other words:
+
+- `A` is a *leaf* node, it has no dependencies (except of course for the input `Aa`)
+- `B` has one dependencies (`A`) plus some input `Ab`
+- `C` has one dependency (`B`) plus some input `Ac`
+
+To simplify things, let's assume every node holds a fetch cached as `refetch` (that is: multiple semi-concurrent requests will reuse a single async request, but requesting the fetch again later on will cause a new refetch).
+
+We'll describe what happens in terms of calls to `query` and `invalidate`, assuming we already have the graph described above obtained with `make()`. We thus omit passing the first argument `graph` in these calls.
+
+We'll also assume that every fetch is performing an async authenticated request to a web server, and thus it needs a `token`.
+`B` and `C` also need something more (they have a dependency on `A` after all): we can imagine this to be whatever, e.g. `C` fetches a user's "posts" and thus it needs the currently authenticated user id, something that can be obtained by `A` that fetches the current authenticated user given a `token`.
```

Comment:
maybe you could add a short paragraph where you give semantic names as example.

an example of what I mean:
```
... we can imagine this to be whatever. This might be an example:
- `A` fetches the current authenticated user
- `B` fetches current user's "posts". In order to work it needs the `Id` of the current user.
- `C` fetches current user's "friends". In order to work it needs the `Id` of the current user.
```